plugins/com.google.cloud.tools.eclipse.googleapis/src/com/google/cloud/tools/eclipse/googleapis/internal/GoogleApiFactory.java

```diff
@@ -0,0 +1,208 @@
+    }
+  };
+
+  public GoogleApiFactory() {
+    this(new GoogleApiUrls());
+  }
+
+  @VisibleForTesting
+  GoogleApiFactory(GoogleApiUrls googleApiUrls) {
+    this.googleApiUrls = googleApiUrls;
+  }
+
+  @Activate
+  public void init() {
+    jsonFactory = new JacksonFactory();
+    buildTransports();
+  }
+
+  /**
+   * @return the CloudResourceManager/Projects API
+   */
+  @Override
+  public Projects newProjectsApi(Credential credential) {
+    synchronized (transports) {
+      HttpTransport transport = transports.get(googleApiUrls.cloudResourceManagerUrl());
+      Preconditions.checkNotNull(jsonFactory, "jsonFactory is null");
+      Preconditions.checkNotNull(transport, "transport is null");
+
+      CloudResourceManager resourceManager =
+          new CloudResourceManager.Builder(transports.get(googleApiUrls.cloudResourceManagerUrl()),
+                                           jsonFactory, credential)
+              .setApplicationName(CloudToolsInfo.USER_AGENT).build();
+      Projects projects = resourceManager.projects();
+      return projects;
+    }
+  }
+
+  /**
+   * @return the Appengine/Apps API
+   */
+  @Override
+  public Apps newAppsApi(Credential credential) {
+    synchronized (transports) {
+      HttpTransport transport = transports.get(googleApiUrls.appEngineAdminUrl());
+      Preconditions.checkNotNull(jsonFactory, "jsonFactory is null");
+      Preconditions.checkNotNull(transport, "transport is null");
+
+      Appengine appengine =
+          new Appengine.Builder(transport, jsonFactory,
+                                credential)
+              .setApplicationName(CloudToolsInfo.USER_AGENT).build();
+      Apps apps = appengine.apps();
+      return apps;
+    }
+  }
+
+  private void buildTransports() {
+    synchronized (transports) {
+      try {
+        buildAppEngineAdminTransport();
+        buildCloudResourceManagerTransport();
+      } catch (URISyntaxException ex) {
+        logger.log(Level.SEVERE, "Could not create transport using the proxy settings", ex);
+      }
+    }
+  }
+
+  private void buildAppEngineAdminTransport() throws URISyntaxException {
+    buildTransportForUrl(googleApiUrls.appEngineAdminUrl());
+  }
+
+  private void buildCloudResourceManagerTransport() throws URISyntaxException {
+    buildTransportForUrl(googleApiUrls.cloudResourceManagerUrl());
+  }
+
+  private void buildTransportForUrl(String url) throws URISyntaxException {
+    ConnectionFactory connectionFactory =
+        new TimeoutAwareConnectionFactory(createProxy(url), DEFAULT_TIMEOUT_MS, DEFAULT_TIMEOUT_MS);
+    transports.put(url,
+                   new NetHttpTransport.Builder().setConnectionFactory(connectionFactory).build());
+  }
+
+  private Proxy createProxy(String url) throws URISyntaxException {
+    Preconditions.checkNotNull(googleApiUrls, "googleApiUrl is null");
+    Preconditions.checkArgument(!url.startsWith("http://"), "http is not supported schema");
+
+    if (proxyService == null) {
```

Comment:
I'm sorry for suggesting you make this dynamic.  You should operate on a copy of the proxyService reference as it may be unbound before the call to `select()` below.